source/brailleDisplayDrivers/seikantk.py

```diff
@@ -0,0 +1,264 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2012-2020 NV Access Limited, Ulf Beckmann <beckmann@flusoft.de>
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+#
+# This file represents the braille display driver for
+# Seika Notetaker, a product from Nippon Telesoft
+# see www.seika-braille.com for more details
+# 29.06.2020 / 12:36
+
+from io import BytesIO
+from typing import List
+import braille
+import brailleInput
+import inputCore
+import hwPortUtils
+import bdDetect
+import hwIo
+from serial.win32 import INVALID_HANDLE_VALUE
+from logHandler import log
+
+TIMEOUT = 0.2
+
+DOT_1 = 0x1
+DOT_2 = 0x2
+DOT_3 = 0x4
+DOT_4 = 0x8
+DOT_5 = 0x10
+DOT_6 = 0x20
+DOT_7 = 0x40
+DOT_8 = 0x80
+
+
+_keyNames = {
+	0x000001: "BACKSPACE",
+	0x000002: "SPACE",
+	0x000004: "LB",
+	0x000008: "RB",
+	0x000010: "LJ_CENTER",
+	0x000020: "LJ_LEFT",
+	0x000040: "LJ_RIGHT",
+	0x000080: "LJ_UP",
+	0x000100: "LJ_DOWN",
+	0x000200: "RJ_CENTER",
+	0x000400: "RJ_LEFT",
+	0x000800: "RJ_RIGHT",
+	0x001000: "RJ_UP",
+	0x002000: "RJ_DOWN"
+}
+
+SEIKA_REQUEST_INFO = b"\x03\xff\xff\xa1"
+SEIKA_INFO = b"\xff\xff\xa2"
+SEIKA_SEND_TEXT = b"\x2c\xff\xff\xa3"
+SEIKA_ROUTING = b"\xff\xff\xa4"
+SEIKA_KEYS = b"\xff\xff\xa6"
+SEIKA_KEYS_ROU = b"\xff\xff\xa8"
+
+SEIKA_CONFIG = b"\x50\x00\x00\x25\x80\x00\x00\x03\x00"
+SEIKA_CMD_ON = b"\x41\x01"
+
+vidpid = "VID_10C4&PID_EA80"
+hidvidpid = "HID\\VID_10C4&PID_EA80"
+SEIKA_NAME = "seikantk"
+
+_dotNames = {}
+for i in range(1, 9):
+	key = globals()["DOT_%d" % i]
+	_dotNames[key] = "d%d" % i
+bdDetect.addUsbDevices(SEIKA_NAME, bdDetect.KEY_HID, {vidpid, })
+
+
+class BrailleDisplayDriver(braille.BrailleDisplayDriver):
+	_dev: hwIo.IoBase
+	name = SEIKA_NAME
+	# Translators: Name of a braille display.
+	description = _("Seika Notetaker")
+	path = ""
+	isThreadSafe = True
+	for d in hwPortUtils.listHidDevices():
+		if d["hardwareID"].startswith(hidvidpid):
+			path = d["devicePath"]
+
+	@classmethod
+	def check(cls):
+		return True
+
+	@classmethod
+	def getManualPorts(cls):
+		return cls.path
+
+	def __init__(self, port="hid"):
+		super(BrailleDisplayDriver, self).__init__()
+		self.numCells = 0
+		self.numBtns = 0
+		self.status = 0
+		self.cmdlen = 0
+		self.handle = None
+		self._hidBuffer = b""
+		log.info("path: " + self.path)
+
+		if self.path == "":
+			raise RuntimeError("No MINI-SEIKA display found, no path found")
+		self._dev = hwIo.Hid(path=self.path, onReceive=self._onReceive)
+		if self._dev._file == INVALID_HANDLE_VALUE:
+			raise RuntimeError("No MINI-SEIKA display found, open error")
+		self._dev.setFeature(SEIKA_CONFIG)  # baudrate, stopbit usw
```

Comment:
This might be an ignorant question, but what does usw mean in this context?

```suggestion
		self._dev.setFeature(SEIKA_CONFIG)  # baud rate, stop bit usw
```

Reply:
I would also like to know. I assume `SEIKA_CONFIG = b"\x50\x00\x00\x25\x80\x00\x00\x03\x00"` is a concatenation of baud rate, stop bit, and usw. It would be nice to know how many bytes for each and what USW stands for.